src/rgw/rgw_sal_motr.cc

```diff
@@ -2650,7 +2650,10 @@ int MotrMultipartUpload::list_parts(const DoutPrefixProvider *dpp, CephContext *
 				     int *next_marker, bool *truncated,
 				     bool assume_unsorted)
 {
-  int rc;
+  int rc = 0;
+  if (num_parts <= 0 or marker < 0)
```

Comment:
JFYI: in this function, we will never get a -ve value since rados has a lower bound of 0, so it will convert any negative value to 0. The only problem why we are getting an error/crash is because we are trying to instantiate the vectors with 0 size which raises an exception.